aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -824,15 +839,108 @@ internal class RealAWSCognitoAuthPlugin(
         onSuccess: Consumer<MutableMap<AuthUserAttributeKey, AuthUpdateAttributeResult>>,
         onError: Consumer<AuthException>
     ) {
-        TODO("Not yet implemented")
+        GlobalScope.launch {
+            try {
+                val userAttributesOptions = options as? AWSCognitoAuthUpdateUserAttributesOptions
+                onSuccess.accept(updateUserAttributes(attributes, userAttributesOptions?.metadata))
+            } catch (e: Exception) {
+                onError.accept(CognitoAuthExceptionConverter.lookup(e, e.toString()))
+            }
+        }
     }
 
     override fun updateUserAttributes(
         attributes: MutableList<AuthUserAttribute>,
         onSuccess: Consumer<MutableMap<AuthUserAttributeKey, AuthUpdateAttributeResult>>,
         onError: Consumer<AuthException>
     ) {
-        TODO("Not yet implemented")
+        updateUserAttributes(attributes, AuthUpdateUserAttributesOptions.defaults(), onSuccess, onError)
+    }
+
+    private suspend fun updateUserAttributes(
+        attributes: MutableList<AuthUserAttribute>,
+        userAttributesOptionsMetadata: Map<String, String>?,
+    ): MutableMap<AuthUserAttributeKey, AuthUpdateAttributeResult> {
+
+        return suspendCoroutine { continuation ->
```

Comment:
Since we are passing success and error callbacks, this function does not a return type.

Reply:
So `suspendCoroutine` also can be removed.